powerline_npm.lua

```diff
@@ -0,0 +1,72 @@
+-- Constants
+
+local function get_package_json_dir(path)
+
+  -- return parent path for specified entry (either file or directory)
+  local function pathname(path)
+          local prefix = ""
+          local postfix = ""
+          local i = path:find("[\\/:][^\\/:]*$")
+          if i then
+                  prefix = path:sub(1, i-1)
+
+          end
+          return prefix
+  end
+
+  if not path or path == '.' then path = clink.get_cwd() end
+
+  local parent_path = pathname(path)
+  return io.open(path..'\\package.json') or (parent_path ~= path and get_package_json_dir(parent_path) or nil)
+end
+
+-- * Segment object with these properties:
+---- * isNeeded: sepcifies whether a segment should be added or not. For example: no Git segment is needed in a non-git folder
+---- * text
+---- * textColor: Use one of the color constants. Ex: colorWhite
+---- * fillColor: Use one of the color constants. Ex: colorBlue
+local segment = {
+  isNeeded = false,
+  text = "",
+  textColor = 0,
+  fillColor = 0
+}
+
+---
+-- Sets the properties of the Segment object, and prepares for a segment to be added
+---
+local function init()
+  segment.isNeeded = get_package_json_dir()
+  if segment.isNeeded then
+    local package_info = segment.isNeeded:read('*a')
+    segment.isNeeded:close()
+
+    local package_name = string.match(package_info, '"name"%s*:%s*"(%g-)"')
+    if package_name == nil then
+            package_version = ''
+    end
+
+    local package_version = string.match(package_info, '"version"%s*:%s*"(.-)"')
+    if package_version == nil then
+            package_version = ''
+    end
+
+    segment.textColor = colorWhite
+    segment.fillColor = colorCyan
```

Comment:
how do we customize this more easily? I can execute clink and this lua under Hyper or Cmder and they each have their own color theme systems

Reply:
I ve actually tried moving that to config, but it has to be processed after the core and throws errors. Not proficient enough in lua to work around it ;p (and to tell the truth I didnt really look hard)

That said, you can change your color directly in cmder settings can you. And I would actually say this should be probably preferable way to change color themes anyway.

Reply:
what kind of customizations are you thinking of?
The order of files does affect results. The only ways to force order I can find (but not 100% sure) are:
- file names: this is why the config file starts with an underscore, to be loaded before all other files
- The number passed to the _register_filter_ function. The higher the number, the later the function is called. This way the folder name can always be first.

Reply:
I was thinking about actually putting colors into the config. As cmder can handle those extra color codes, but then we could put it in core as well. So i dont know if it is even worth it to extract that.